crates/typos-cli/src/config.rs

```diff
@@ -30,6 +45,11 @@ impl Config {
 
     pub fn from_file(path: &std::path::Path) -> Result<Self, anyhow::Error> {
         let s = std::fs::read_to_string(path)?;
+
+        if path.file_name().unwrap() == "pyproject.toml" {
+            return Ok(toml::from_str::<PyprojectTomlConfig>(&s)?.tool.typos);
+        }
+
         Self::from_toml(&s)
```

Comment:
Wrote up some thoughts in #361.  I originally thought we were going to need dicussion beyond this PR but I think we can handle it within this PR.

If `pyproject.toml` does not have a `tool.typos` table, we should keep looking for a config.

To do this, we'll need
- `pub typos: Config` to be `pub typos: Option<Config>`
- Either `from_file` to now return `Option<Self>` or for a new `from_file_` that returns `Option<Self>` with `from_file` doing a `unwrap_or_default`

Reply:
Sorry for the delay on this. I just pushed a commit that should address this.

One thing I ran into is that `find_project_file` is called before `from_file`, so it is hard to figure out what the next config file should be. I fixed this by making `find_project_file` return a list of files instead, that way `from_dir` can iterate over each `from_file` result and handle the potential `None` type.

Actually, I'm just now realizing I moved `pyproject.toml` to the front of `SUPPORTED_FILE_NAMES` for testing purposes then moved it to the back before pushing, so this looping logic might not be needed anymore. If we happen to rearrange `SUPPORTED_FILE_NAMES` though, or want to support more language-specific files in the future, then the for loop should probably stay. What are your thoughts?

Reply:
> If we happen to rearrange SUPPORTED_FILE_NAMES though, or want to support more language-specific files in the future, then the for loop should probably stay. What are your thoughts?

Yes, it seems appropriate for it to stay